### PR TITLE
Update developer documentation

### PR DESCRIPTION
**Pull request by lamblin**
Mention Blocks and Fuel, as the rules should be valid for both repositories.

### CODE REVIEW

File: docs/development/index.rst [hunk 1]
```diff
@@ -210,9 +218,11 @@ the error is being raised for.
 
 Unit testing
 ------------
-Blocks uses unit testing to ensure that individual parts of the library behave
-as intended. It's also essential in ensuring that parts of the library are not
-broken by proposed changes.
+Blocks and Fuel use unit testing to ensure that individual parts of
+the library behave as intended. It's also essential in ensuring that
+parts of the library are not broken by proposed changes. Since Blocks
+depends on Fuel, it is important to make sure changes in Fuel do not
```
Comment (dmitriy-serdyuk) [inline docs/development/index.rst]: It doesn't really depend on Fuel. I would say they designed to be used together.

**Comment by rizar** [comment]
@lamblin , do you intend to finish this?

**Comment by lamblin** [comment]
Updated, sorry about the delay.

File: docs/development/index.rst [hunk 2]
```diff
@@ -143,16 +149,18 @@ Not doing so `clobbers the original traceback`_, making it impossible to use
 
 Serialization
 ~~~~~~~~~~~~~
-To ensure the reproducibility of scientific experiments Blocks tries to make
-sure that stopping and resuming training doesn't affect the final results. In
-order to do so it takes a radical approach, serializing the entire training
-state using pickle_. Some things cannot be pickled, so their use should be
-avoided when the object will be pickled as part of the main loop:
+To ensure the reproducibility of scientific experiments, Blocks and Fuel
```
Comment (rizar) [inline docs/development/index.rst]: This paragraph is about Blocks only, I guess. The pronoun "it" can not refer to Blocks and Fuel.

**Comment by rizar** [comment]
LGTM, except for the pickling-related paragraph. A bit of rephrasing is needed there.

**Comment by rizar** [comment]
I looked at the single paragraph that I had not like and found it not too bad. This PR has been around for a while, time to merge :)

**Pull request merged**